libs/runtime/ebpf_platform.c

```diff
@@ -513,12 +515,21 @@ ebpf_allocate_preemptible_work_item(
     return ebpf_result_from_cxplat_status(status);
 }
 
+#define CPU_INFO_PROCESSOR_INFO_FUNCTION 1
+#define CPU_INFO_SSE42_BYTE_OFFSET 2
+#define CPU_INFO_SSE32_BIT_OFFSET 20
+
 _Must_inspect_result_ ebpf_result_t
 ebpf_platform_initiate()
 {
     ebpf_result_t result = ebpf_result_from_cxplat_status(cxplat_initialize());
     _ebpf_platform_is_cxplat_initialized = (result == EBPF_SUCCESS);
     ebpf_initialize_cpu_count();
+    // Check if processor supports SSE4.2
+    int cpu_info[4] = {0};
+    __cpuid(cpu_info, CPU_INFO_PROCESSOR_INFO_FUNCTION);
```

Comment:
Maybe it is premature to worry about arm64, but should ebpf abstract this processor-specific intrinsic into a separate header/file or use `#ifdef _M_X64` compile guards here?

Reply:
That will be taken care of as part of the ARM64 bring up, which will occur once there is ARM64 github runner support.